pkg/api/config.go

```diff
@@ -81,6 +81,15 @@ type Config struct {
 
 	MasterStartupSASURI string `json:"-"`
 	WorkerStartupSASURI string `json:"-"`
+
+	// Private cluster configs, not exposed to the end user
+
+	// PrivateEndpoint contains IP or DNS of PrivateEndpoint
+	// If this field is not nil, value should be used to reach
+	// api server. Otherwise, use public API DNS.
+	PrivateEndpoint                  string `json:"-"`
```

Comment:
need to check naming on these fields, and I don't think they should be in `type Config struct`

Reply:
Would be good do decide where? NetworkProfile?

Reply:
networkProfile where we should move this